errors.go

```diff
@@ -8,6 +8,12 @@ type FlagsmithAPIError struct {
 	msg string
 }
 
+type FlagsmithErrorHandler struct {
```

Comment:
This type is not really an error handler, it's more of an error object. Would something like `FlagsmithRequestError` make more sense?

Also, do we need to introduce a new type at all, or could we add the response code to the existing `FlagsmithAPIError` type?

Reply:
I'm okay with this. In that case, struct will look like 
```
type FlagsmithAPIError struct {
	Msg                              string
	Err                                 error
	ResponseStatusCode int
	ResponseStatus          string
}
```
@gagantrivedi Let me know if you prefer this

Reply:
Why do we need both `ResponseStatusCode` and `ResponseStatus` here?

Reply:
Yeah, I'd prefer that